crypto/signature.go

```diff
@@ -83,18 +84,25 @@ func (x *Signature) Calculate(signer Signer, data []byte) error {
 // CalculateMarshalled signs data using Signer and encodes public key for subsequent verification.
 // If signer is a StaticSigner, just sets prepared signature.
 //
+// Pre-allocated byte slice can be passed in buf parameter to avoid new allocations. In ideal case buf length should be
+// StableSize length. If buffer length shorter than StableSize or nil, new slice will be allocated.
+//
 // Signer MUST NOT be nil.
 //
 // See also Verify.
-func (x *Signature) CalculateMarshalled(signer Signer, obj StablyMarshallable) error {
+func (x *Signature) CalculateMarshalled(signer Signer, obj StablyMarshallable, buf []byte) error {
```

Comment:
Looks like no one outside of SDK uses this API directly, so even though I don't like changes like this at RC11 stage, likely we can accept it.